engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java

```diff
@@ -574,6 +580,11 @@ public VolumeInfo createVolumeFromSnapshot(Volume volume, Snapshot snapshot, Use
             throw e;
         }
 
+        boolean kvmIncrementalSnapshot = SnapshotManager.kvmIncrementalSnapshot.valueIn(_hostDao.findClusterIdByVolumeInfo(snapInfo.getBaseVolume()));
+        if (kvmIncrementalSnapshot && DataStoreRole.Image.equals(dataStoreRole)) {
+            snapInfo = snapshotHelper.convertSnapshotIfNeeded(snapInfo);
```

Comment:
does this convert existing non-incremental snapshot to incremental snapshot while creating volume from it?

Reply:
No, we have no way of converting existing snapshots from non-incremental (AKA full) to incremental snapshots. This will only convert incremental snapshots to full ones in order to create volumes from it.